index.bs

```diff
@@ -2690,23 +2690,15 @@ partial interface MLGraphBuilder {
 ### The slice() method ### {#api-mlgraphbuilder-slice}
 Produce a slice of the input tensor.
 <script type=idl>
-dictionary MLSliceOptions {
-  sequence<unsigned long> axes;
-};
-
 partial interface MLGraphBuilder {
-  MLOperand slice(MLOperand input, sequence<long> starts, sequence<long> sizes,
-                optional MLSliceOptions options = {});
+  MLOperand slice(MLOperand input, sequence<unsigned long> starts, sequence<unsigned long> sizes);
 };
 </script>
 <div algorithm=slice>
     **Arguments:**
         - *input*: an {{MLOperand}}. The input tensor.
-        - *starts*: a sequence of {{long}}. The starting indices to slice of the corresponding axes of the input shape. A negative index value is interpreted as counting back from the end. For example, the value -1
-        - *sizes*: a sequence of {{long}}. The lengths to slice of the corresponding axes of the input shape.
-            The length value of -1 selects all the remaining elements from the starting index of the given axis.
-        - *options*: an optional {{MLSliceOptions}}. The optional parameters of the operation.
-            - *axes*: a sequence of {{unsigned long}}. The dimensions of the input shape to which *starts* and *sizes* apply. The values in the sequence must be in the range [0, N-1] where N is the rank of input tensor. When not specified, the sequence is assumed to be [0, ..., N-1], e.g. [0,1,2] for a 3-D tensor.
+        - *starts*: a sequence of {{unsigned long}}. The sequence of unsigned integer values indicating the starting index to slice of each input dimension, of length N where N is the rank of the input tensor. For each dimension *d* of *input*, *starts[d]* indicates the starting index to slice in that dimension.
```

Comment:
Does it need to describe explicitly that **starts index** for each dimension d must be in the range **[0, input size - 1]**?
@huningxin @fdwr WDYT? Thanks.

Reply:
The *starting index* may not be *input size - 1*, because there would be 0 elements to slice which is not allowed in the current definition.

The *starting index* should also satisfy the constraint *starting index + size <= input size* where *size* must not be 0 for each dimension.

Later, as @fdwr [recommended](https://github.com/webmachinelearning/webnn/pull/376#discussion_r1162180789), if 0 slice size is allowed, this constraint should also work.

@BruceDai , do you see any challenges to implement it?

Reply:
> do you see any challenges to implement it?

No. My concern is only for adding explicitly description for starting index.

Reply:
> The starting index may not be input size - 1

The starting index could be input size - 1  when size = 1.
For example producing a slice of the 1D input tensor( it's value is [0, 1, 2, 3] ) by starts = [3] and sizes = [1],  result is expected as [3], and now starting index + size = 4 = input size satisfy **the constraint starting index + size <= input size**. Am I understanding this correctly? PTAL, thanks.

Reply:
Thanks @BruceDai , your example is correct. My previous thought was wrong (comment updated). I'd agree it makes sense to add [0, input size - 1] constraint to starting index.

Reply:
Fixed in the latest commit https://github.com/webmachinelearning/webnn/pull/376/commits/8efc5685cc9adea78debe3cb3d2e829afda7331a, please take a look. Thanks!

Reply:
LGTM, thanks.

Reply:
> The starting index may not be input size - 1

What I thought was if we allow 0 size dimension for the use cases that @fdwr [mentioned ](https://github.com/webmachinelearning/webnn/pull/376#discussion_r1162180789), the range would become [0, -1] which won't make sense. 

Because we will discuss that in a separate issue, it would be fine to add starting index range [0, input size - 1] into this PR. Any thoughts @fdwr ?

Reply:
@huningxin : Yes, given the current minimum requirement of size > 0, then `[0, inputSize - 1]` is the correct range.